packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -1970,6 +2003,18 @@ const configureStore = (options: CrudStoreOptions & GridStoreOptions) => {
       void store.refreshDocuments();
     });
 
+    globalAppRegistry.on(
```

Comment:
This is still incorrect and causes an issue that triggers the modal to show up on all currently open collection tabs:

![Kapture 2023-11-14 at 14 43 14](https://github.com/mongodb-js/compass/assets/5036933/4c9dd707-81e0-4f49-ba80-8546bff9cc16)

This event should be scoped to the `localAppRegistry`, not global one

Reply:
As mentioned, it's between different plugins, so the localAppRegistry doesn't work, unless there is some configuration that is not set up properly, which I don't see anywhere. 

Probably the correct solution would be the tab to ignore the event if it's not visible, which is common behaviour in lots of applications. Is there a way to detect the visible tab on the event handler?

Reply:
The `localAppRegistry` is _specifically_ a mechanism for different plugins to communicate inside one scope. For plugins registered under `Collection.Tab` role, the localAppRegistry scope is defined. This is how `open-create-index-modal`, `open-*-export-to-language`, `open-explain-plan-modal` events work

Reply:
Can you clarify a bit what's not working for you? I pulled your branch locally and just replaced globalAppRegistry for the emit and on for this event to localAppRegistry with no other changes and it works as expected

Reply:
I think I was doing something wrong when I tested it, as mentioned in the standup. I tried again and it worked flawlessly, as you mentioned, so I'm going to commit the change ❤️ .